portfolio/src/main/java/com/google/sps/servlets/ViewPuzzlePageServlet.java

```diff
@@ -0,0 +1,26 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles requests sent to the /viewpuzzle URL.*/
+@WebServlet("/viewpuzzle")
+public class ViewPuzzlePageServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+
+    try {
+      Long puzzleId = Long.parseLong(request.getParameter("id"));
+      response.sendRedirect("viewpuzzle.html?id=" + puzzleId.toString());
+    } catch(NumberFormatException e) {
+      //the id couldn't be parsed
```

Comment:
Set a response code and message indicating that the request was malformed:
```suggestion
     response.setStatus(HttpServletResponse. SC_BAD_REQUEST);
     response.getWriter().println("Invalid puzzleId");
```